pkg/controllers/cspi-controller/handler.go

```diff
@@ -215,7 +228,7 @@ func (c *CStorPoolInstanceController) updateStatus(cspi *cstor.CStorPoolInstance
 
 	// Since we queried in following order health and io.openebs:readonly output also
 	// will be in same order
-	valueList, err := zpool.GetListOfPropertyValues(pool, propertyList)
+	valueList, err := zpool.GetListOfPropertyValues(pool, propertyList, c.zcmdExecutor)
```

Comment:
We can have `GetListOfPropertyValues` as a method? This will make life easier. Open to discussion.

Reply:
We can take this one later.

Reply:
But -- Imagine next time we want to customise this guy  `GetListOfPropertyValues` we need to change a lot of places the argument signature. 
Having it as a method of an struct just will make the life easy.

Reply:
Realise this with the existing command that is being used 
```
oc.SetCompression()
```
SetCompression() is bound to a struct and see that we did not require a change in arguments at all the place but the job will be done in 2 simple steps :
1. You can extend the struct bound to SetCompression by adding fields.
2. Implement the new fields in the new SetCompression method. 
Voila...
No more changes...

Reply:
So we can follow the SetCompression strategy in property listing also..

Reply:
> We can take this one later.

Let me take this comment in this PR itself

Reply:
I also have the same idea having as a method instead of a function but proceed with this because if someone wants to use this then they no need to worry about OperationConfig they can call this function but It is good to have as amethod